birdhouse/scripts/read-configs.include.sh

```diff
@@ -166,19 +177,40 @@ read_components_default_env() {
 
     # EXTRA_CONF_DIRS and DEFAULT_CONF_DIRS relative paths are relative to COMPOSE_DIR.
     if [ -d "$COMPOSE_DIR" ]; then
-        cd "$COMPOSE_DIR"
+        cd "$COMPOSE_DIR" >/dev/null
     fi
 
     source_conf_files "$DEFAULT_CONF_DIRS" 'DEFAULT_CONF_DIRS'
     source_conf_files "$EXTRA_CONF_DIRS" 'EXTRA_CONF_DIRS'
 
     # Return to previous pwd.
     if [ -d "$COMPOSE_DIR" ]; then
-        cd -
+        cd - >/dev/null
     fi
 }
 
 
+check_optional_vars() {
+
+    for i in ${OPTIONAL_VARS}
+    do
+        v="${i}"
+        d=`eval echo "$v"`
```

Comment:
Is there a reason for backticks instead of `$()`?

Also, is the reason we have to do the `eval` because we have to resolve the delayed eval variables? If so, why not just call `check_optional_vars` after `process_delayed_eval` ?

Reply:
The backticks here is for the compatibility of `eval` mentioned by @tlvu.
I do not want to enter too much into the handling resolution of the vars in the context of this PR. Maybe there is another way to do it avoiding eval, but that should be done in another PR.

Reply:
Yes indeed, I was not able to get `eval` working without backticks.